File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>

// Strict compiler function declaration
float get_positive_float(void);

int main(void)
{
    // Get the amount owed in positive numbers only
    float owed = get_positive_float();
    
    // Keep track of the number of used coins
    int counter = 0;
    
    // Find the type of used coins
    while (owed > 0)
    {
        // Subtract the value of the largest coin type until 'owed' is zero
        if (owed >= 0.25)
        {
            owed -= 0.25;
        }
        else if (owed >= 0.1)
        {
            owed -= 0.1;
        }
        else if (owed >= 0.05)
        {
            owed -= 0.05;
        }
        else
        {
            owed -= 0.01;
        }
        
        // One coin was used
        counter++;
    }
    
    printf("%i\n", counter);
}

float get_positive_float(void)
{
    float money;
    
    // Reprompt the user until they enter a positive decimal
    do
    {
        money = get_float("Change owed: ");
    }
    while (money < 0);
    
    return money;
}

File: pset6/cash/cash.py
from cs50 import get_float


def main():
    # Get the amount owed in positive rounded integers
    owed = get_positive_rounded()

    # Define avaliable coins and keep track of their total used number
    coins = [25, 10, 5, 1]
    counter = 0
    
    # Find the type of used coins
    while owed > 0:
        # Subtract the value of the largest coin type until 'owed' is zero
        for coin in coins:
            if int(owed / coin) >= 1:
                owed -= coin
                break

        # One coin was used
        counter += 1
    
    print(counter)


def get_positive_rounded():
    while True:
        # Reprompt the user until they enter a positive decimal
        money = get_float('Change owed: ')
        if money > 0:
            break
    
    return round(money * 100)


if __name__ == '__main__':
    main()